staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -486,3 +541,244 @@ func getObservedGeneration(obj *unstructured.Unstructured, condition map[string]
 	statusObservedGeneration, found, _ := unstructured.NestedInt64(obj.Object, "status", "observedGeneration")
 	return statusObservedGeneration, found
 }
+
+// JSONPathWait holds a JSONPath Parser which has the ability
+// to check for the JSONPath condition and compare with the API server provided JSON output.
+type JSONPathWait struct {
+	jsonPathExpression string
```

Comment:
This would be used for the test cases because I didn't take the input for the user
```go
	o := &WaitOptions{
				ResourceFinder: genericclioptions.NewSimpleFakeResourceFinder(test.infos...),
				DynamicClient:  fakeClient,
				Timeout:        test.timeout,

				Printer: printers.NewDiscardingPrinter(),
				ConditionFn: JSONPathWait{
					jsonPathExpression: test.jsonPathExp,
					jsonPathCondition:  test.jsonPathCond,
					jsonPathParser:     j, errOut: ioutil.Discard}.IsJSONPathConditionMet,
				IOStreams: genericclioptions.NewTestIOStreamsDiscard(),
			}

```

Reply:
Hmm, I don't understand yet. 

It looks like `jsonPathParser` is already created with `jtest.sonPathExp` here:

```
j, _ := newJSONPathParser(test.jsonPathExp)
```

And then `j` is used as the value for `jsonPathParser` on `JSONPathWait` construction:
```
                                   ConditionFn: JSONPathWait{
					jsonPathExpression: test.jsonPathExp,
					jsonPathCondition:  test.jsonPathCond,
					jsonPathParser:     j,
                                        errOut:             ioutil.Discard}.IsJSONPathConditionMet,
```

So why does `JSONPathWait` also need you to set `jsonPathExpression`?  Doesn't the parser already contain everything it needs to find the matches?

Reply:
Oops, I got it now and deleted the expression